reusable_components_nlp/README.md

```diff
@@ -0,0 +1,989 @@
+
+# End-to-end Reusable ML Pipeline with Seldon and Kubeflow
+
+In this example we showcase how to build re-usable components to build an ML pipeline that can be trained and deployed at scale.
+
+We will automate content moderation on the Reddit comments in /r/science building a machine learning NLP model with the following components:
+
+![](img/completed-pipeline-deploy.jpg)
+
+This tutorial will break down in the following sections:
+
+1) Run all the services (Kubeflow and Seldon)
+
+2) Test and build all our reusable pipeline steps
+
+3) Use Kubeflow to Train the Pipeline and Deploy to Seldon
+
+5) Test Seldon Deployed ML REST Endpoints
+
+6) Visualise Seldon's Production ML Pipelines
+
+# Before you start
+Make sure you install the following dependencies, as they are critical for this example to work:
+
+* Helm v2.13.1+
+* A Kubernetes cluster running v1.13 or above (minkube / docker-for-windows work well if enough RAM)
+* kubectl v1.14+
+* ksonnet v0.13.1+
+* kfctl 0.5.1 - Please use this exact version as there are major changes every few months
+* Python 3.6+
+* Python DEV requirements (we'll install them below)
+
+Let's get started! 🚀🔥 We will be building the end-to-end pipeline below:
+
+![](img/kubeflow-seldon-nlp-full.jpg)
+
+
+
+```python
+!cat requirements-dev.txt
+```
+
+    python-dateutil
+    https://storage.googleapis.com/ml-pipeline/release/0.1.20/kfp.tar.gz
+    kubernetes
+    click
+    seldon_core
+    numpy
+
+
+
+```python
+!pip install -r requirements-dev.txt
+```
+
+## 1) Run all the services (Kubeflow and Seldon)
```

Comment:
This could use some clarity that it will install default Kubeflow components on an existing cluster. Include an escape hatch for users that already have KF installed.

Reply:
I don't understand what you mean. What is an escape hatch? Perhaps if there are other examples that have this outlined I could try to add the same terminology when they install kubeflow. It should be quite self explanatory tho, if they already have KF installed they should skip the installation of Kubeflow.

Reply:
By escape hatch I just mean a brief statement that this section is optional if kubeflow is already installed on the cluster (the section refers to running the services, which is a bit ambiguous for new users). For users unfamiliar with `kfctl`, it is unclear which steps generate files and which ones modify the cluster. Being a bit more explicit about what is going on makes it smoother for new users.